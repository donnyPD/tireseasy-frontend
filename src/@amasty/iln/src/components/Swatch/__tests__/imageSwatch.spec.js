import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import ImageSwatch from '../imageSwatch';
import Icon from '@magento/venia-ui/lib/components/Icon';

jest.mock('@magento/venia-ui/lib/components/Icon', () => 'Icon');
jest.mock('@magento/peregrine/lib/util/imageUtils', () => {
  return {
    generateUrl: jest.fn(() => () => '')
  };
});

const defaultProps = {
  swatchData: '#ccc',
  label: 'Test',
  isSelected: false
};

test('renders a Swatch correctly', () => {
  const component = createTestInstance(<ImageSwatch {...defaultProps} />);

  expect(() => component.root.findByType(Icon)).toThrow();

  expect(component.toJSON()).toMatchSnapshot();
});

test('renders a Swatch correctly image', () => {
  const props = {
    ...defaultProps,
    swatchData: '/imgSrc'
  };
  const component = createTestInstance(<ImageSwatch {...props} />);

  expect(component.toJSON()).toMatchSnapshot();
});

test('renders an icon if isSelected is true', () => {
  const props = {
    ...defaultProps,
    isSelected: true
  };

  const component = createTestInstance(<ImageSwatch {...props} />);

  expect(() => component.root.findByType(Icon)).not.toThrow();
  expect(component.toJSON()).toMatchSnapshot();
});
