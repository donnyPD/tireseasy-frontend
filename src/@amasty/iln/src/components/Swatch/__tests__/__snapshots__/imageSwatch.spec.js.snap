// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Swatch correctly 1`] = `
<span
  style={
    Object {
      "--venia-swatch-bg": "#ccc",
    }
  }
>
  <span>
    <span
      style={
        Object {
          "--venia-swatch-bg": "#ccc",
        }
      }
    />
    <span>
      Test
    </span>
  </span>
</span>
`;

exports[`renders a Swatch correctly image 1`] = `
<span
  style={
    Object {
      "--venia-swatch-bg": "url(\\"\\")",
    }
  }
>
  <span>
    <span
      style={
        Object {
          "--venia-swatch-bg": "url(\\"\\")",
        }
      }
    />
    <span>
      Test
    </span>
  </span>
</span>
`;

exports[`renders an icon if isSelected is true 1`] = `
<span
  style={
    Object {
      "--venia-swatch-bg": "#ccc",
    }
  }
>
  <Icon
    src={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "color": [Function],
          "size": [Function],
        },
        "render": [Function],
      }
    }
  />
  <span>
    <span
      style={
        Object {
          "--venia-swatch-bg": "#ccc",
        }
      }
    />
    <span>
      Test
    </span>
  </span>
</span>
`;
